plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -168,7 +171,10 @@ export const CatalogTable = (props: CatalogTableProps) => {
     },
   ];
 
-  const currentKind = filters.kind?.value || '';
+  let currentKind = filters.kind?.value || '';
+  if (!loadingKinds && currentKind) {
+    currentKind = allKinds[currentKind] || currentKind;
+  }
```

Comment:
the problem here is that the `useAllKinds()` hook results in a loading state which means there's two titles that get renders, for example `Owned components (1)` and then `Owned Components (1)`. Does anyone know the best way to deal with this?